SConstruct

```diff
@@ -1644,11 +1644,18 @@ def doConfigure(myenv):
         if not conf.CheckLib("execinfo"):
             Exit(1)
 
+    # check for presence of timegm(3) and polyfill if needed
     conf.env['MONGO_HAVE_TIMEGM'] = conf.CheckDeclaration(
         'timegm', includes="#include <time.h>", language='C')
     if conf.env['MONGO_HAVE_TIMEGM']:
         conf.env.Append(CPPDEFINES=['MONGO_HAVE_TIMEGM'])
 
+    # check for presence of strnlen(3) and polyfill if needed
+    conf.env['MONGO_HAVE_STRNLEN'] = conf.CheckDeclaration(
```

Comment:
I think this needs to be done as a config.h entry, rather than a CPPDEFINE fs we ever call strnlen from a header, and it appears that we do in bsonelement.h.

Have a look at how MONGO_HAVE_HEADER_UNISTD_H is handled.